src/Solve.h

```diff
@@ -14,9 +14,10 @@ namespace Internal {
  * tree and place it as far to the left as possible, and as far up the
  * tree as possible (i.e. outside most parentheses). If the expression
  * is an equality or comparison, this 'solves' the equation. Returns
```

Comment:
... this 'solves' the equation. Returns a pair<bool, Expr>. The Expr is the mutated expression, and the bool indicates whether there is a single instance of the variable in the result. If it is false, the expression has only been partially solved, and there are still multiple instances of the variable.